### ISSUE TITLE
GetVendors refresh time potentially inaccurate?

### ISSUE THREAD

**Issue by Cytraen** [Bungie-net/api#1401]
Currently, I'm grabbing vendor inventories (via GetVendors with `Vendors` and `VendorSales` components) right at reset, and checking if their `nextRefresh` or if any `overrideNextRefresh` times are before the current UTC time, and if they are, I throw out the response and wait another minute before grabbing them again. However, I'm getting updated refresh times with outdated inventories.

Am I doing something wrong? Is the API never going to return an outdated `nextRefresh` because it assumes the data provided by the game servers is always up-to-date?

I would prefer not comparing the previous inventory to the current one, in the rare-but-still-possible case the few inventory items I care about don't change.

**Comment by jshaffstall-bng** [comment]
This sounds like it could be Bungie.net-side 'D2 account state' caching behavior. Can you provide before-and-after JSON responses from a repro?

Then we can file a bug and look into it.

**Comment by ashakoor** [comment]
Alright, I set up a client to make a request every ~5 seconds, starting from `16:53:50.04Z`
All the responses I got were identical up to and including the one at `16:59:59.28Z`, they all the correct data for Jan 31st.
The first request after reset at `17:00:04.21Z` had only the `nextRefresh` and `overrideNextRefresh` times changed vs. the previous response, no hashes or anything were changed, only refresh times.
The next minute of responses were identical, then the one at `17:01:03.64Z` was the first with the inventories correctly updated for Feb 1st, and the ones for the next minute were also identical.

Posted the full JSON responses to Pastebin:

[`16:53:50.04Z` / Jan 31st](https://pastebin.com/xamMkZ3L)
[`17:00:04.21Z` / Inventory-Refresh mismatch](https://pastebin.com/D4e3siZJ)
[`17:01:03.64Z` / Feb 1st](https://pastebin.com/Qm374EH9)

Member Type: `3` / `TigerSteam`
Member ID: `4611686018511231764`
Character ID: `2305843009750375960`
Components: `Vendors` and `VendorSales` / `400` and `402`

I will also note the refreshTimes changing right after 17:00:00Z is not always true, I've had them change just before (like 16:59:58Z after the response is parsed) as well, though that may just be my system's time not being exactly in sync with the server's.
